### PR TITLE
Remove fallback to jwks when fetching jwk

### PR DESCRIPTION
**Pull request by j3parker**
* Everyone supports JWK
* The fallback can cause perf issues due to the response size of jwks, its uncachability and a bug with one of the public key hosts in the system
* Let's keep everyone supporting jwk

### CODE REVIEW

File: src/D2L.Security.OAuth2.TestFramework/AuthServiceMock.cs [hunk 1]
```diff
@@ -94,7 +94,7 @@ public enum KeyType {
 				keyDtos.Add( dto );
 
 				m_server
-					.Stub( r => r.Get( $"jwk/{ key.Id }" ) )
+					.Stub( r => r.Get( $"/jwk/{ key.Id }" ) )
```
Comment (j3parker) [inline src/D2L.Security.OAuth2.TestFramework/AuthServiceMock.cs]: This was broken - but tests were falling back to jwks :)

File: src/D2L.Security.OAuth2/Keys/Default/Data/JwksProvider.cs [hunk 2]
```diff
@@ -39,18 +39,8 @@ Uri authEndpoint
...
-					// route isn't supported which will make this recover (but
-					// slowely.) Where it matters (the LMS and auth) JWK is
-					// supported so this shouldn't be necessary. We don't
-					// expect many "legitimate" 404s (keys that don't exist.)
-					// so in practice this should only happen when it's
-					// actually important, if it ever happens.
-					if( res.StatusCode != HttpStatusCode.OK ) {
-						return await ( this as IJwksProvider ).RequestJwksAsync().SafeAsync();
+					if ( res.StatusCode == HttpStatusCode.NotFound ) {
+						return JsonWebKeySet.Empty( url );
```
Comment (j3parker) [inline src/D2L.Security.OAuth2/Keys/Default/Data/JwksProvider.cs]: This interface returns a `JsonWebKeySet`, usually a single key. To keep the exception messages correct it was sanest to return an empty JWKS here, and let the thing that notices that and throws `PublicKeyNotFoundException` do its thing.

File: src/D2L.Security.OAuth2/Keys/Default/JsonWebKeySet.cs [hunk 3]
```diff
@@ -34,9 +34,19 @@ internal sealed class JsonWebKeySet {
...
-			m_keys = ImmutableArray.Create( jsonWebKey );
+		internal JsonWebKeySet( JsonWebKey jsonWebKey, Uri src )
+			: this(
+				ImmutableArray.Create( jsonWebKey ),
+				src
+			) { }
+
+		private JsonWebKeySet( ImmutableArray<JsonWebKey> keys, Uri src ) {
+			m_keys = keys;
+			Source = src ?? throw new NotImplementedException( nameof( src ) );
```
Comment (travisMiehm) [inline src/D2L.Security.OAuth2/Keys/Default/JsonWebKeySet.cs]: `ArgumentNullException`?

Comment (j3parker) [reply src/D2L.Security.OAuth2/Keys/Default/JsonWebKeySet.cs]: Oh shit thanks, auto-complete fail

**Comment by j3parker** [comment]
> Verified okay for existed auth service clients?

Yep, a bit ago, but I'll revalidate today.

> Also, we'll need to make a branch and cherry-pick this over, master isn't shippable (it is theoretically, but not practically)

D'oh, ok. Do you know what commit we should branch at?

**Comment by omsmith** [comment]
Likely candidates:
* 5f7e2b410bcb5794e3a5942b5edac78e534c6ea8 (7.2.2)
* a85271cffd78d4cb4036fb34949c587e430947fd (7.2.2 formatting / code cleanups)

**Comment by j3parker** [comment]
I'll go with a85271c to avoid possible merge conflicts

**Comment by omsmith** [comment]
Made a branch [7.x](https://github.com/Brightspace/D2L.Security.OAuth2/commits/7.x) with https://github.com/Brightspace/D2L.Security.OAuth2/commit/58d78092d914ac5f76affde9b364825054f263b8 cherry-picked. These apply cleanly on top.

**Comment by j3parker** [comment]
I can cherry-pick this tomorrow and we'll figure out how exactly we want to do this.

**Pull request merged**